tools/lib/h5diff_array.c

```diff
@@ -3364,48 +3364,66 @@ print_pos(diff_opt_t *opts, hsize_t idx, size_t u)
             hsize_t curr_pos = idx;
 
             parallel_print("[ ");
-            H5TOOLS_DEBUG("do calc_acc_pos[%ld] nelmts:%d - errstat:%d", i, opts->hs_nelmts, opts->err_stat);
+            H5TOOLS_DEBUG("do calc_acc_pos[%ld] nelmts:%d - errstat:%d", idx, opts->hs_nelmts,
+                          opts->err_stat);
 
             if (opts->sset[0] != NULL) {
                 /* Subsetting is used - calculate total position */
-                hsize_t elmnt_cnt = 1;
-                hsize_t dim_cnt   = 0;   /* previous dim size */
-                hsize_t str_cnt   = 0;   /* previous dim stride */
-                hsize_t curr_idx  = idx; /* calculated running position */
-                hsize_t str_idx   = 0;
-                hsize_t blk_idx   = 0;
-                hsize_t cnt_idx   = 0;
-                hsize_t hs_idx    = 0;
-                j                 = opts->rank - 1;
+                hsize_t prev_dim_size = 0; /* previous dim size */
```

Comment:
What's this change about?

Reply:
This is the code to convert a hyperslab contiguous index to the original index coordinates for the display output.